src/main/scala/Joint.scala

```diff
@@ -47,14 +113,28 @@ object Joint {
   * @param shape A list of lines specifying the shape of the joint. Each item is a
   * 3-tuple. The first two items specify the line, while the last gives the distance
   * of the line from the joint's center in the local coordinate system.
+  * @param dipAngleParam An optional parameter that can be used to specify the dip angle for the joint.
+  *                 This avoids recalculation of a known dip angle.
+  * @param dipDirectionParam An optional parameter that can be used to specify the dip direction for
+  *                     the joint. This avoids recalculation of a known dip direction.
   */
 case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Double, Double),
-                 center: (Double, Double, Double), dipAngle: Double, dipDirection: Double, phi: Double,
-                 cohesion: Double, shape: Seq[((Double, Double, Double),Double)]) {
+                 center: (Double, Double, Double), phi: Double, cohesion: Double,
+                 shape: Seq[((Double, Double, Double),Double)],
+                 dipAngleParam: Option[Double]=null,
```

Comment:
Okay, let's do something like the following:
```scala
case class Joint(..., dipAngleParam: Option[Double]=None, ...) {
    ...
    val dipAngle = dipAngleParam match {
        case None => Joint.dipAngle(normalVec)
        case Some(da) => da
    }
...
}
```
I was using `null` before because the member variable itself was an `Option` so I couldn't use `Some` or `None` as a default value. In this case we can use an `Option` with `None` as its default value to indicate that the dip angle should be calculated and otherwise we extract the value from the option. In either case, the dip angle we store is itself a `Double` rather than an `Option`.

How does this sound?